ifaddr/_shared.py

```diff
@@ -69,7 +75,13 @@ class IP(object):
     Represents an IP address of an adapter.
     """
 
-    def __init__(self, ip, network_prefix, nice_name):
+    # Type of an IPv4 address (a string in "xxx.xxx.xxx.xxx" format)
+    IPv4Address = Union[str]
```

Comment:
Let's put those at module level and prefix with `_` to not suggest those are to be used by the client code.